### PR TITLE
feat(issue-details): Customize replay breadcrumbs to show issue and event info

### PR DESCRIPTION
**Pull request by malwilley**
Ref https://github.com/getsentry/sentry/issues/63157

Modifies the breadcrumb items (in tooltips and the sidebar when in fullscreen mode) to include a linkable event ID and issue shortcode. These links are only clickable when going to a different issue/event, otherwise they are just displayed as text. The focused event is shown as "This Event".

This introduces a new context `ReplayGroupContext` which provides the group and event ID. When available, the replay breadcrumbs will highlight the current event/issue. This context is provided by the EventReplay component, but could be used in other places as well down the line.

CleanShot 2024-01-29 at 22 06 50

### CODE REVIEW

**Review by ryan953 — COMMENTED** [review]
Lets change the rendering for all breadcrumbs, this "custom" stuff doesn't need to be special, and its a bunch of poor names anyway.

**Comment by malwilley** [comment]
> Lets change the rendering for all breadcrumbs, this "custom" stuff doesn't need to be special, and its a bunch of poor names anyway.

@ryan953 the problem is that what's displayed in the breadcrumbs is contextual based on which event you're on (showing "This Event" and link differences) so replays would need to know the issue and event IDs. I didn't think we wanted issue-specific logic in here but if you're okay with that I can certainly do it.

**Comment by ryan953** [comment]
@malwilley lets talk about how to architect it, because right now bigs files are getting bigger and the 'snippet' logic should not be mixed in with the logic for 'there is one event to highlight'

File: static/app/components/events/eventReplay/index.spec.tsx [hunk 1]
```diff
@@ -1,48 +1,89 @@
+import merge from 'lodash/merge';
```
Comment (ryan953) [inline static/app/components/events/eventReplay/index.spec.tsx:1]: no mo lo

**Comment by malwilley** [comment]
Instead of using a customization context, this now uses a context that provides eventId and groupId. The new event and issue links are also shown in the normal replays experience as well. Updated the description to reflect that.

File: static/app/components/events/eventReplay/index.spec.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/components/events/eventReplay/index.spec.tsx]: tru

**Review by ryan953 — APPROVED** [review]

**Pull request merged**